### PR TITLE
Use RISC Zero BigInt modular multiplication accelerator

### PR DESCRIPTION
**Pull request by nategraf**
Building on https://github.com/risc0/risc0/pull/466, this PR enables the use of the RISC Zero
256-bit modular multiplication accelerator within guest code.

This PR is based on crypto-bigint@v0.5.2

### CODE REVIEW

**Comment by nategraf** [comment]
NOTE: The `U128::mul_wide` implementation doesn't seem to give significant performance advantage. It's possibly due to the inefficiency of padding the inputs and splitting the result, but this is not clear. I'll spend a few more moments seeing if there is a performance advantage to be realized, but if not, I will remove it from this PR.

File: src/uint/modular.rs
Comment (jbruestle) [inline src/uint/modular.rs]: Seems like this is a pure formatting change, maybe back it out for clarity?

**Comment by nategraf** [comment]
TODO: I just realized that this PR, by representing residues in standard form rather than Montgomery, breaks the existing serialization code if passed between host and guest. I'll need to address this before the PR can be merged.

**Comment by nategraf** [comment]
Following up on https://github.com/risc0/RustCrypto-crypto-bigint/pull/1#issuecomment-1552281130, without specialization or other language support, I don't think we can automatically apply the wide multiplication accelerator mode of 128-bit Uints. I've refactored to drop that, but still expose a function for a `mul_wide` specifically with `U128` inputs.

**Comment by nategraf** [comment]
I've now addressed https://github.com/risc0/RustCrypto-crypto-bigint/pull/1#issuecomment-1561562989 in [e949138](https://github.com/risc0/RustCrypto-crypto-bigint/pull/1/commits/e9491387d0791a0129d9e90481d4582c6686443d)

@jbruestle, this is fully ready to review. Once this is merged, I will also tag the commit and adjust https://github.com/risc0/RustCrypto-elliptic-curves/pull/1 to point to the new tag, at which point it will be ready to merge as well (after a review)

**Review by jbruestle — APPROVED** [review]
Looks good!

File: src/uint/modular.rs
Comment (nategraf) [reply src/uint/modular.rs]: Done

**Pull request merged**
